packages/app-builder-lib/templates/nsis/uninstaller.nsh

```diff
@@ -28,6 +28,109 @@ Function un.onInit
   !endif
 FunctionEnd
 
+Function un.atomicRMDir
```

Comment:
I'm not entirely proud of this, but it implements `RMDir /r` that preserves backups of all files and directories that it removed in `$PLUGINSDIR\old-install`.